price-estimator/src/filter.rs

```diff
@@ -0,0 +1,218 @@
+use crate::orderbook::Orderbook;
+use serde::Serialize;
+use serde_with::rust::display_fromstr;
+use std::num::ParseIntError;
+use std::sync::Arc;
+use warp::{Filter, Rejection};
+
+/// Validate a request of the form
+/// `/markets/<baseTokenId>-<quoteTokenId>/estimated-buy-amount/<sellAmountInQuoteToken>`
+/// and answer it.
+pub fn estimated_buy_amount<T: Send + Sync + 'static>(
+    orderbook: Arc<Orderbook<T>>,
+    price_rounding_buffer: f64,
+) -> impl Filter<Extract = (String,), Error = Rejection> + Clone {
+    estimated_buy_amount_filter().and_then({
+        move |token_pair, sell_amount_in_quote| {
+            let orderbook = orderbook.clone();
+            async move {
+                let orderbook = orderbook.get_reduced_orderbook().await;
+                let result = estimate_buy_amount(
+                    token_pair,
+                    sell_amount_in_quote,
+                    price_rounding_buffer,
+                    orderbook,
+                );
+                // The compiler cannot infer the error type because we never return an error.
```

Comment:
Something that might be subject to change based on the `atoms` as an input parameter discussion?

Reply:
Probably. I'll do the atom thing in another PR we will see how the function changes then.